### DISCUSSION TITLE
Social relay issue with liquidsoap

### DISCUSSION THREAD

**Discussion by wezo3** [savonet/liquidsoap#2459] [General]
Hello Guys,
I have this Error when try to relay to youtube or fackbook

```
At /home/liquidsoap.liq, line 61, char 0-5:
video)
Error 5: this value has type
  string (inferred at /home/liquidsoap.liq, line 58, char 8-50)
but it should be a subtype of
  source(_)

`
audio = "/root/media/sound.mp3"

video = "/root/media/image.jpg"
video = ffmpeg.encode.video(
%ffmpeg(%video(codec="libx264")),
video)

stream = mux_video(video=video, audio)

enc = %ffmpeg(
format="flv",
%audio.copy,
%video.copy)

# Send to YouTube
key = "YOUTUBE_KEY"
url = "rtmp://a.rtmp.youtube.com/live2/#{key}"
output.url(url=url, enc, stream)


# Send to Facebook
key = "FB_KEY"
url = "rtmps://live-api-s.facebook.com:443/rtmp/#{key}"
output.url(self_sync=true, url=url, enc, stream)
```

`

and all requirements installed with opem ffmpeg ocurl ..etc 

what mean this error and how can fix this ?
Thanks

**Comment by wezo3** [comment]
@toots .. also please this issue, we will be happy with your help
Thanks

**Comment by Russsgithub** [comment]
@wezo3 

video = "/root/media/image.jpg"

is a string but should be a source. I don't use video in liq at the moment, but maybe

video = single("/root/media/image.jpg")

may work.

The book and/or docs should explain how to do this.

Russ

**Reply by wezo3** [reply]
Thank you for your replay but still same issue when used video = single("/root/media/image.jpg")
At /home/liquidsoap.liq, line 61, char 0-5:
video)
Error 5: this value has type
  string (inferred at /home/liquidsoap.liq, line 58, char 8-50)
but it should be a subtype of
  source(_)

**Reply by Russsgithub** [reply]
audio is also a string not a aource. ?

The book and docs are very good for liquidsoap.

**Reply by wezo3** [reply]
Thank you it works now after your hint and  after searched about this hint in liquidsoap book as you said 
I appreciate your help

**Comment by Russsgithub** [comment]
No worries. Your welcome.

On 22 Jun 2022 11:01, wezo3 ***@***.***> wrote:

Thank you it works now after your hint and after searched about this hint in liquidsoap book as you said
I appreciate your help
